### PR TITLE
Improve error message when `overrideBootstrapCommand` is required for `customAMIs`

### PR DESCRIPTION
**Pull request by TiberiuGC**
### Description

When using one of AL2 or Ubuntu `customAMIs`, there is a validation in place enforcing the user to specify a bootstrap script.
https://github.com/eksctl-io/eksctl/blob/3ac2b24b6819da5f551d43fcb7764d57fa199f7b/pkg/apis/eksctl.io/v1alpha5/validation.go#L792-L793

However, if the user does not provide the script, the error message currently states that `overrideBootstrapCommand` is always required when using a `customAMI`, which is not really true as some `amiFamilies` such as Bottlerocket and Windows don't support it. Thus the error message may create confusion (see https://github.com/eksctl-io/eksctl/issues/6837#issue-1811791532).

This PR aims to make the error massage more clear.

Related to #6837 

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by a-hilaly — APPROVED** [review]

File: pkg/apis/eksctl.io/v1alpha5/managed_nodegroup_test.go [hunk 1]
```diff
@@ -63,7 +65,7 @@ var _ = Describe("Managed Nodegroup Validation", func() {
 					AMIFamily: DefaultNodeImageFamily,
 				},
 			},
-			errMsg: "overrideBootstrapCommand is required when using a custom AMI",
+			errMsg: fmt.Sprintf("overrideBootstrapCommand is required when using a %s custom AMI", DefaultNodeImageFamily),
```
Comment (cPu1) [inline pkg/apis/eksctl.io/v1alpha5/managed_nodegroup_test.go:68]: For AL2, this will read:

```
overrideBootstrapCommand is required when using a AmazonLinux2 custom AMI
```

instead of

```
overrideBootstrapCommand is required when using an AmazonLinux2 custom AMI
```

Ubuntu also suffers from the same issue. To avoid this, we can reword it to:

```suggestion
			errMsg: fmt.Sprintf("overrideBootstrapCommand is required when using a custom AMI based on %s", DefaultNodeImageFamily),
```

**Review by cPu1 — COMMENTED** [review]
I have left a comment about rewording the error message but otherwise LGTM.

**Review by yuxiang-zhang — APPROVED** [review]

**Review by cPu1 — APPROVED** [review]
